lib/vnet/app_resolver.go

```diff
@@ -169,8 +170,9 @@ func (r *TCPAppResolver) ResolveTCPHandler(ctx context.Context, fqdn string) (*T
 		}
 
 		leafClusterName := ""
-		if clusterClient.ClusterName() != profileName {
-			leafClusterName = clusterClient.ClusterName()
+		clusterName := clusterClient.ClusterName()
+		if clusterName != "" && clusterName != profileName && clusterName != clusterClient.RootClusterName() {
```

Comment:
Just curious: is it possible to `clusterName == profileName` but `clusterName != clusterClient.RootClusterName()`?

Reply:
That would be possible if you have a root cluster with the name `foo` available at `foo.com` and a leaf cluster with the name of `foo.com`.

Which is not very likely, but I suppose to be correct, we should perhaps only compare `ClusterName` to `RootClusterName`.

The problem with the previous implementation was that if the name of the root cluster was not equal to its proxy host (and thus the profile name), VNet would set `leafClusterName` to that of the root cluster. This worked fine in the client cache. However, in methods such as `OnInvalidLocalPort`, we'd generate invalid teleterm URIs, as they'd end up being like `/clusters/<profile name>/leaves/<root cluster name>`.